### DISCUSSION TITLE
What OSes are supported?? (including porting to other OSes using Rust)

### DISCUSSION THREAD

**Discussion by yumnycheesecakes** [ruffle-rs/ruffle#12855] [Q&A] [answered]
As the title reads it :)

**Comment by p0008874** [comment]
https://doc.rust-lang.org/nightly/rustc/platform-support.html

**Comment by Dinnerbone** [comment, marked as answer]
Ruffle Web will run on any browser that supports WASM. It'll run much better if it supports WebGL too, but it doesn't strictly need it.


Ruffle Desktop runs on Linux, Mac and Windows. I don't know the specific version minimums because we haven't tested it, but it'll likely be whatever Rust will support. It requires Direct X 12, Vulkan, Metal or OpenGL to function. Otherwise, not much other dependencies.


The core ruffle code isn't platform specific and has been ported to other places such as Android, too. iOS is theoretically possibly but nobody wants to invest the time to check, given Apple's policies against emulators.

**Reply by yumnycheesecakes** [reply]
This is good, at least for iOS we have the browser way. If there is Rust Toolchain for a platform, then we can port Ruffle on it, right??
